File: src/test/java/nz/co/searchwellington/controllers/RssUrlBuilderTest.java
package nz.co.searchwellington.controllers;

import static org.junit.Assert.assertEquals;
import nz.co.searchwellington.model.SiteInformation;
import nz.co.searchwellington.model.Tag;

import org.junit.Before;
import org.junit.Test;
import org.mockito.Mock;
import org.mockito.Mockito;
import org.mockito.MockitoAnnotations;

public class RssUrlBuilderTest {

	@Mock SiteInformation siteInformation;
	@Mock Tag tag;

	@Before
	public void setup() {
		MockitoAnnotations.initMocks(this);
		Mockito.when(siteInformation.getAreaname()).thenReturn("Somewhere");
	}
	
	@Test
	public void rssDescriptionForTagShouldBeSetFromTagDescriptionTextIfAvailable() throws Exception {
		RssUrlBuilder rssUrlBuilder = new RssUrlBuilder(siteInformation);
		
		Mockito.when(tag.getDescription()).thenReturn("This is a tag about something...");
		
		assertEquals(tag.getDescription(), rssUrlBuilder.getRssDescriptionForTag(tag));
	}
	
}

File: src/test/scala/nz/co/searchwellington/controllers/RssUrlBuilderTest.scala
package nz.co.searchwellington.controllers

import java.util.UUID

import nz.co.searchwellington.model.{SiteInformation, Tag}
import org.junit.Assert.assertEquals
import org.junit.Test

class RssUrlBuilderTest {

  private val siteInformation = new SiteInformation(areaname = "Somewhere", url = "https://wellynews.local")
  private val rssUrlBuilder = new RssUrlBuilder(siteInformation)

  @Test
  def rssDescriptionForTagShouldBeSetFromTagDescriptionTextIfAvailable(): Unit = {
    val tag = Tag(id = UUID.randomUUID().toString, description = Some("This is a tag about something..."))

    assertEquals("This is a tag about something...", rssUrlBuilder.getRssDescriptionForTag(tag))
  }

  @Test
  def tagRssUrlIsBasedOnTagName(): Unit = {
    val tag = Tag(id = UUID.randomUUID().toString, description = Some("This is a tag about something..."), name = "atag")

    assertEquals("https://wellynews.local/atag/rss", rssUrlBuilder.getRssUrlForTag(tag))
  }

}
